### PR TITLE
[READY] Adds pytest support

### LINKED ISSUE #273: Create simple pytest framework for ansible inventory
## Description

https://github.com/socallinuxexpo/scale-network/blob/master/ansible/inventory.py has become an important glue for the majority of provisioning and inquiries into the provisioned hardware. We should take the necessary steps to add a `pytest` framework to start validating on the parsing of our inventory.

## Acceptance Criteria
* Added framework for `pytest` and `tox`
* Integrated that into the CI test suite
* Spun out follow up issues as needed

### PR DESCRIPTION
**Pull request by kylerisse**
## Description of PR
Fixes #273 

Adds pytest to the circle workflow and unit tests for inventory.py

## Previous Behavior
No unit tests for inventory.py

## New Behavior
Unit tests for inventory.py integrated into CI process

## Tests
This github PR in WAIT state

### CODE REVIEW

**Comment by kylerisse** [comment]
Renamed the circle job ansible_lint to ansible_test to more accurately reflect its new functionality. I don't know if that breaks CI, but it seems to have caused circle to be stuck waiting for a now non-existent ansible_lint.

**Comment by sarcasticadmin** [comment]
@kylerisse I had to remove the `required` from the test you renamed so that all open PRs dont hang in a check wait status. Anyway, once this is merged Ill rebase my branch and then we can enable the require for `ansible_test`.

File: ansible/test_inventory.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        ["VLAN\tBadVLAN\t\tABC\t2001:470:f325:504::/64\t10.128.4.0/24\tBad VLAN", None],
+    ]
+    for line, vlan in cases:
+        assert inventory.makevlan(line, "Conference") == vlan, line
+
+
+def test_bitmasktonetmask():
+    '''test cases for the bitmasktonetmask() function'''
+    cases = [
+        [16, None],
```
[the remark this answers is not in the document]
Comment (kylerisse) [reply ansible/test_inventory.py]: Correct, I coded the tests to the existing behavior. It's an assumption. Adding tests has shed some light on very poor implementations. Now that we have them I'll have the confidence to refactor this entire endeavor.

**Comment by kylerisse** [comment]
Thanks @sarcasticadmin . I have a few more tests to add before I want to call it done.

File: .circleci/config.yml [hunk 2]
```diff
@@ -74,33 +74,29 @@ jobs:
       - store_artifacts:
           path: /tmp/ipq806x
 
-  ansible_lint:
+  ansible_tests:
     docker:
-    - image: circleci/python:3.7.2-node-browsers-legacy
-    resource_class: medium
+    - image: kylerisse/ansible-tester@sha256:f6a9507ec1d7a2dd0570cd88cb19746840c4f97e7618f53158616b9e1f5a3618
```
[the remark this answers is not in the document]
Comment (kylerisse) [reply .circleci/config.yml]: responsible container management. I have the container builds integrated into Docker Hub so they can be audited at any time.

File: ansible/inventory.py [hunk 3]
```diff
@@ -116,11 +109,11 @@ def genvlans(line, building):
...
 
 
-def populatevlans():
-    '''populate the vlan list'''
+def populatevlans(vlansdirectory, vlansfile):
+    '''populate the vlan list from a vlans diretory and file'''
     vlans = []
     # seed root file
-    todo = [["#include\tvlans", "vlan"]]
+    todo = [["#include\t" + vlansfile + "\tno_building"]]
```
[the remark this answers is not in the document]
Comment (kylerisse) [reply ansible/inventory.py]: to make the functions testable i had to change them to take file names as arguments vs using the globals directly. I also named all test files differently than their real name, to weed out any assumptions. That exercise exposed this.

File: ansible/test_inventory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sarcasticadmin) [reply ansible/test_inventory.py]: 100% agree with the approach, I just wanted to make sure I was following this correctly

**Comment by sarcasticadmin** [comment]
@kylerisse thanks for getting those last tests added, this all looks great! Another plus is our Ansible/Python CI tests went from 1 min to a few seconds with that new container! :shipit:

**Pull request merged**